File: sdk/node/src/yaml.js
'use strict';
/**
 * YAML parsing for profile.yaml files.
 *
 * Thin wrapper around `js-yaml` (a runtime dependency). The previous
 * hand-rolled parser was retired after a diff against js-yaml exposed two
 * latent bugs: it did not strip inline `#` comments (so `streaming: false # x`
 * became the string `"false # x"` and the runner treated streaming as on),
 * and it rendered an empty `env:` value as `""` instead of `null`.
 *
 * `parseYamlSimple` is kept as an alias for backwards compatibility with
 * callers that imported it by that name.
 */

const yaml = require('js-yaml');

function parseYaml(text) {
  // js-yaml handles JSON input too (JSON is a YAML subset), so the explicit
  // JSON.parse fast-path the hand-rolled parser had is unnecessary.
  return yaml.load(text);
}

module.exports = { parseYaml, parseYamlSimple: parseYaml };

File: sdk/python/src/agentproc/yaml.py
"""YAML parsing for profile.yaml files.

Thin wrapper around PyYAML (a runtime dependency). The previous hand-rolled
subset parser lived in ``cli.py`` and silently mis-parsed several constructs
that real profiles use — most notably inline ``#`` comments, so
``streaming: false # one-shot mode`` became the string ``"false # one-shot mode"``
and the runner's ``is not False`` check left streaming **on**. That is the same
class of bug that retired the Node SDK's hand-rolled parser in favour of
``js-yaml``; the Python SDK now matches.

``parse_yaml_simple`` is kept as an alias for any caller that imported it by
that name.
"""

from __future__ import annotations

from typing import Any, Dict

import yaml


def parse_yaml(text: str) -> Dict[str, Any]:
    """Parse a YAML document into a Python object.

    Uses ``yaml.safe_load``. JSON input is a YAML subset, so the explicit
    JSON fast-path the hand-rolled parser had is unnecessary.
    """
    return yaml.safe_load(text)


# Backwards-compat alias.
parse_yaml_simple = parse_yaml


__all__ = ["parse_yaml", "parse_yaml_simple"]
